csharp_dotnetcore/14.NLP-With-Dispatch/NlpDispatch/AppInsights/MyAppInsightsLuisRecognizer.cs

```diff
@@ -0,0 +1,124 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.ApplicationInsights;
+using Microsoft.Bot.Builder;
+using Microsoft.Bot.Builder.AI.Luis;
+using Newtonsoft.Json.Linq;
+
+namespace NLP_With_Dispatch_Bot.AppInsights
+{
+    /// <summary>
+    /// MyAppInsightLuisRecognizer invokes the Luis Recognizer and logs some results into Application Insights.
+    /// Logs the Top Intent, Sentiment (label/score), (Optionally) Original Text, Conversation and ActivityID.
+    /// The Custom Event name this logs is <see cref="MyLuisConstants.IntentPrefix"/> + "." + 'found intent name'
+    /// For example, if intent name was "add_calender":
+    ///    LuisIntent.add_calendar
+    /// See <see cref="LuisRecognizer"/> for additional information.
+    /// </summary>
+    public class MyAppInsightLuisRecognizer : LuisRecognizer
+    {
+        /// <summary>
+        /// Initializes a new instance of the <see cref="MyAppInsightLuisRecognizer"/> class.
+        /// </summary>
+        /// <param name="application">The LUIS _application to use to recognize text.</param>
+        /// <param name="predictionOptions">The LUIS prediction options to use.</param>
+        /// <param name="includeApiResults">TRUE to include raw LUIS API response.</param>
+        public MyAppInsightLuisRecognizer(LuisApplication application, LuisPredictionOptions predictionOptions = null, bool includeApiResults = false, bool logOriginalMessage = false, bool logUserName = false)
+            : base(application, predictionOptions, includeApiResults)
+        {
+            LogOriginalMessage = logOriginalMessage;
+            LogUsername = logUserName;
+        }
+
+        /// <summary>
+        /// Gets a value indicating whether to log the <see cref="Activity"/> message text that came from the user.
+        /// </summary>
+        /// <value>If true, will log the Activity Message text into the AppInsight Custome Event for Luis intents.</value>
+        public bool LogOriginalMessage { get; }
+
+        /// <summary>
+        /// Gets a value indicating whether to log the User name.
```

Comment:
>User [](start = 55, length = 4)

user

Reply:
Removing app insights, this is going away